nvbench/CMakeLists.txt

```diff
@@ -24,7 +24,12 @@ set(srcs
   detail/measure_cold.cu
   detail/measure_hot.cu
   detail/state_generator.cxx
-)
+
+  detail/measure_cupti.cuh
```

Comment:
Rather than unconditionally compiling these and slicing bits of them out with `NVBENCH_HAS_CUPTI`, let's do this:

```
if (NVBench_ENABLE_CUPTI)
  list(APPEND srcs [cupti files here])
endif()
```

and then remove the `NVBENCH_HAS_CUPTI` checks from these files. This clean separation will reduce the complexity of maintaining the optional dependency, since the only significant code divergence will be in `state_exec.cuh`.

We'll need to move error logging for the "cupti is not enabled but requested" case in `state_exec.cuh` with this approach.

Reply:
Don't list the header files here -- it's harmless, but currently we're only listing translation units in the `srcs` list.